sqlglot/generator.py

```diff
@@ -2721,10 +2721,13 @@ def trycast_sql(self, expression: exp.TryCast) -> str:
         return self.cast_sql(expression, safe_prefix="TRY_")
 
     def log_sql(self, expression: exp.Log) -> str:
```

Comment:
why is this changing?

Reply:
Because it relies on the order of `expression.args` which is incorrect. For example if you do something like `log = exp.Log(expression=..., this=...)` then the arg order won't match the order in arg types and we'll get unexpected behavior. In the context of this PR, the second test I added initially failed because of this.